UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -584,7 +586,15 @@ void SendInfoToBrain()
             }
 
             info.memories = action.memories;
-            info.storedVectorActions = action.vectorActions;
+            if(done)
```

Comment:
I fail to see how this solves the problem, could you add some comments explaining what problem it solves and how ?

Reply:
I added a short comment - my initial description on the pull request itself was wrong. Since Reset() is called before SendInfoToBrain(), when Reset happens, action.vectorActions is zeroed out, then SendInfoToBrain() is called and the 0's are stored in storedVectorActions. This is OK for training since the Python code doesn't use the storedVectorActions. (now that I think about it, we might need to do this for the textActions as well.)

I don't neccessarily like this fix as it isn't very clean. But it seems like the only thing possible without reordering the Academy step loop.

Reply:
Should we not zero out action.vectorActions at reset ?

Reply:
Is there a reason we do this? Maybe we can initialize them to 0 on creation but not zero them out? I guess it's problematic since 0 isn't null, it has an actual meaning when it comes to actions